store/sqlstore/reaction_store.go

```diff
@@ -110,22 +112,31 @@ func (s *SqlReactionStore) BulkGetForPosts(postIds []string) ([]*model.Reaction,
 
 func (s *SqlReactionStore) DeleteAllWithEmojiName(emojiName string) error {
```

Comment:
This function is only executed when you remove the Custom emoji, so... should it be "permanent delete" instead of mark as deleted? Maybe we can rename it to "PermanentDeleteAllWithEmojiName" to be explicit about the behavior, but I think this should keep permanent deleting the reactions, because the underneath custom emoji is no longer there.

Reply:
Yes, I was unaware that was the only use of that method.  It should be renamed.  Still, soft delete here does no harm since there is no way to undelete since the emoji no longer exists and the soft deleted record will never appear in any queries.

Reply:
Thinking about this some more...  I think perhaps `DeleteAllWithEmojiName` should do a soft delete.  Since we will be keeping reaction history it should be complete.  It can be used for compliance/audit to prove someone read a post.  Despite the emoji not longer existing the records are safe in that they can never be undeleted, unless the emoji is re-added.

Reply:
Sounds sensible. I think now the reactions are not included in the compliance export, but something to consider, of course.